packages/svast/index.d.ts

```diff
@@ -0,0 +1,95 @@
+import { Node, Position, Point, Parent, Data, Literal } from 'unist';
+export { Node, Position, Point, Parent, Data, Literal };
+
+// TODO: script and style tags don't have nodes yet
+
+export interface SvelteParent extends Parent {
+	children: (
+		| SvelteElement
+		| SvelteComponent
+		| Comment
+		| Text
+		| SvelteExpression
+		| VoidBlock
+		| BranchingBlock
+		| EachBlock
+		| SvelteTag
+	)[];
+}
+
+export interface Root extends Parent {
+	type: 'root';
+}
+
+export interface BaseSvelteTag extends SvelteParent {
+	tagName: string;
+	properties: (Property | Directive)[];
+	selfClosing: boolean;
+}
+
+export interface SvelteTag extends BaseSvelteTag {
+	type: 'svelteTag';
+}
+
+export interface SvelteElement extends BaseSvelteTag {
+	type: 'svelteElement';
+}
+
+export interface SvelteComponent extends BaseSvelteTag {
+	type: 'svelteComponent';
+}
+
+export interface BaseProperty extends Node {
+	name: string;
+	shorthand: 'none' | 'boolean' | 'expression';
+	value: (Text | SvelteExpression)[];
+	modifiers: Literal[];
+}
+
+export interface Property extends BaseProperty {
+	type: 'svelteProperty';
+}
+
+export interface Directive extends BaseProperty {
```

Comment:
Same with these guys. Would just add extra information whenever needed.

```ts
export type Directive = BaseProperty<'svelteDirective'> & { specifier: string };
// or 
export interface Directive extends BaseProperty<'svelteDirective'> { 
  specifier: string;
}
```
I prefer the 2nd but both options still read well 🤷